### PR TITLE
disulfide bonds read in xml

### PR DESCRIPTION
**Pull request by lschaffer2**
need this info for samples where we didn't reduce (insulin)

### CODE REVIEW

File: UsefulProteomicsDatabases/ProteinDbLoader.cs [hunk 1]
```diff
@@ -309,6 +315,11 @@ public static List<Protein> LoadProteinXML<T>(string proteinDbLocation, bool onT
                                                     decoyBigPeptideTypes[oneBasedBeginPositions.Count - i - 1] = peptideTypes[i];
                                                 }
 
+                                                foreach (DisulfideBond disulfideBond in disulfideBonds)
+                                                {
+                                                    decoy_disulfides.Add(new DisulfideBond(reversed_sequence.Length - disulfideBond.OneBasedBeginPosition + 1, reversed_sequence.Length - disulfideBond.OneBasedEndPosition + 1, disulfideBond.Description));
```
Comment (stefanks) [inline UsefulProteomicsDatabases/ProteinDbLoader.cs]: Can you motivate this computation?

File: UsefulProteomicsDatabases/ProteinDbLoader.cs [hunk 2]
```diff
@@ -242,6 +243,10 @@ public static List<Protein> LoadProteinXML<T>(string proteinDbLocation, bool onT
                                             else if (oneBasedfeature_position >= 1)
                                                 sequenceVariations.Add(new SequenceVariation(oneBasedfeature_position, original_value, variation_value, feature_description));
                                         }
+                                        else if (feature_type == "disulfide bond")
+                                        {
+                                            disulfideBonds.Add(new DisulfideBond((int)oneBasedbeginPosition, (int)oneBasedendPosition, feature_description));
```
Comment (stefanks) [inline UsefulProteomicsDatabases/ProteinDbLoader.cs]: Can oneBasedbeginPosition or oneBasedendPosition ever be null?

Comment (lschaffer2) [inline UsefulProteomicsDatabases/ProteinDbLoader.cs]: Thanks for catching this - yes. I'll account for this.

File: UsefulProteomicsDatabases/ProteinDbLoader.cs (hunk 1 above)
Comment (lschaffer2) [inline UsefulProteomicsDatabases/ProteinDbLoader.cs]: I thought for decoys the disulfides should be at the same cysteine positions in the reverse sequence

Comment (stefanks) [inline UsefulProteomicsDatabases/ProteinDbLoader.cs]: Could you add this to a unit test?

**Pull request merged**